File: src/main/java/com/github/snowgooseyk/sscsv/base/WriteBuffer.java
package com.github.snowgooseyk.sscsv.base;

import java.io.Closeable;
import java.io.Flushable;

public interface WriteBuffer extends Closeable, Flushable {

	WriteBuffer append(int index, byte[] bites);

	WriteBuffer append(int index, char[] charactors);

	WriteBuffer append(int index, String value);

	WriteBuffer scroll();

    int getCurrentIndex();
}

File: src/main/scala/com/github/snowgooseyk/sscsv/WriteBuffer.scala
package com.github.snowgooseyk.sscsv

import com.github.snowgooseyk.sscsv.base.{ WriteBuffer => JWriteBuffer }
import com.github.snowgooseyk.sscsv.base.dsv.DelimitedWriteBuffer

/**
 * @author snowgooseyk
 */
case class WriteBuffer(underlying: JWriteBuffer, index: Int = 0) {
  def ++(value: String) = {
    underlying.append(1, value)
    copy
  }
  def !() = {
    underlying.flush
    copy
  }
  def ln() = {
    underlying.scroll
    copy
  }
  private[this] def copy = WriteBuffer(underlying, index + 1)
}
